service/src/main/java/bio/terra/axonserver/service/cloud/gcp/CloudStorageService.java

```diff
@@ -1,20 +1,50 @@
 package bio.terra.axonserver.service.cloud.gcp;
 
+import bio.terra.axonserver.utils.BoundedByteArrayOutputStream;
 import com.google.auth.oauth2.GoogleCredentials;
-import com.google.cloud.storage.Blob;
+import com.google.cloud.ReadChannel;
+import com.google.cloud.storage.BlobId;
 import com.google.cloud.storage.StorageOptions;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import javax.ws.rs.InternalServerErrorException;
 
+/** Service for interacting with Google Cloud Storage */
 public class CloudStorageService {
+  private static final int MAX_OBJECT_SIZE = 512 * 1024 * 1024; // 512 MB
 
   public CloudStorageService() {}
 
-  public Blob getBucketObject(
+  /**
+   * Get the contents of a GCS bucket object
+   *
+   * @param googleCredentials Google credentials to use for the request
+   * @param bucketName Name of the bucket
+   * @param objectName Name of the object
+   * @return Contents of the object
+   */
+  public byte[] getBucketObject(
       GoogleCredentials googleCredentials, String bucketName, String objectName) {
 
-    return StorageOptions.newBuilder()
-        .setCredentials(googleCredentials)
-        .build()
-        .getService()
-        .get(bucketName, objectName);
+    BlobId blobId = BlobId.of(bucketName, objectName);
+    try (ReadChannel reader =
+        StorageOptions.newBuilder()
+            .setCredentials(googleCredentials)
+            .build()
+            .getService()
+            .reader(blobId)) {
+
+      BoundedByteArrayOutputStream outputStream = new BoundedByteArrayOutputStream(MAX_OBJECT_SIZE);
```

Comment:
why bounded? why are we restricting the bucket object size?

Reply:
Dan pointed out the case where a user could try to fetch a very large file and cause out of memory exceptions. For the current use case of jupyter notebooks and R markdown, these files wouldn't be more than 20-30mb